packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -443,26 +501,33 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
     }
 
     private getTimeForX(x: number): bigint {
-        if (!this.lineChartRef.current?.chartInstance) {
+        if (!this.chartRef.current?.chartInstance) {
             return BigInt(0);
         }
         const offset = this.props.viewRange.getOffset() ?? BigInt(0);
         const duration = this.props.viewRange.getDuration();
         const time = this.props.viewRange.getStart() - offset +
-            BIMath.round(x / this.lineChartRef.current.chartInstance.width * Number(duration));
+            BIMath.round(x / this.chartRef.current.chartInstance.width * Number(duration));
         return time;
     }
 
     protected getXForTime(time: bigint): number {
-        if (!this.lineChartRef.current?.chartInstance) {
+        if (!this.chartRef.current?.chartInstance) {
             return 0;
         }
         const start = this.props.viewRange.getStart();
         const duration = this.props.viewRange.getDuration();
-        const x = Number(time - start) / Number(duration) * this.lineChartRef.current.chartInstance.width;
+        const x = Number(time - start) / Number(duration) * this.chartRef.current.chartInstance.width;
         return x;
     }
 
+    protected getOffset(): bigint {
+        // Work around loss of precision when working with BigInt
+        let offset = this.props.viewRange.getOffset() ?? BigInt(0);
```

Comment:
The view range's offset is very dangerous to use because it depends how it was constructed. It shouldn't even be API...

TimeRange(123, 456, 0), TimeRange(0, 333, 123) and TimeRange(23, 356, 100) all represent the same time range but with different offset.

Long story short, we should be using viewRange.getStart() instead.

Reply:
Replaced with getStart().